package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -450,15 +509,15 @@ def __init__(self, atomgroup, **kwargs):
         self.atomgroup = atomgroup
         residues = atomgroup.residues
         protein = atomgroup.universe.select_atoms("protein").residues
```

Comment:
```suggestion
        protein = atomgroup.select_atoms("protein").residues
```

I would expect the analysis to be limited to whichever atom group I passed in.